src/mongoc/mongoc-client.c

```diff
@@ -1783,8 +1783,20 @@ _mongoc_client_command_with_opts (mongoc_client_t *client,
       if ((mode & MONGOC_CMD_WRITE) &&
           !mongoc_write_concern_is_default (default_wc) &&
           (!opts || !bson_has_field (opts, "writeConcern"))) {
-         bool is_fam =
-            !strcasecmp (_mongoc_get_command_name (command), "findandmodify");
+         const char *command_name;
+         bool is_fam;
+
+         command_name = _mongoc_get_command_name (command);
```

Comment:
I suggest putting this check first in the function to do the simple check early. Otherwise we risk the same bug again later if we need the command name in a different branch of this function.

Reply:
Can do. That certainly improves maintainability. I didn't start with that since it entailed adding overhead to the entire function.

Reply:
OK. Checking whether the command name is NULL (meaning the command is empty) is cheap constant time, let's go ahead and do it. Checking if the command name is "findandmodify" could conceivably take a couple microseconds, on the other hand.